internal/wallet/utxo.go

```diff
@@ -2,12 +2,40 @@ package wallet
 
 import (
 	"github.com/btcsuite/btcd/btcjson"
+	"github.com/btcsuite/btcwallet/walletdb"
 )
 
 // Utxo is a unspend transaction output
-type Utxo btcjson.ListUnspentResult
+type Utxo = btcjson.ListUnspentResult
 
-// ListUnspent returns unspent transactions
-func (w *wallet) ListUnspent() (utxos []Utxo, err error) {
-	return
+// ListUnspent returns unspent transactions.
+// TODO: add filter
+//   Only utxos with address contained the param addresses will be considered.
+//   If param addresses is empty, all addresses are considered and there is no
+//   filter
+func (w *wallet) ListUnspent() (utxos []*Utxo, err error) {
+	var results []*btcjson.ListUnspentResult
+	err = walletdb.View(w.db, func(tx walletdb.ReadTx) error {
+		addrmgrNs := tx.ReadBucket(waddrmgrNamespaceKey)
+		txmgrNs := tx.ReadBucket(wtxmgrNamespaceKey)
+
+		syncBlock := w.manager.SyncedTo()
```

Comment:
Does the waddmgr do some block downloading task?
I though it just manages addresses and priv/pub keys of them.